src/main/java/org/oasis_eu/portal/core/dao/impl/CatalogStoreImpl.java

```diff
@@ -214,7 +214,9 @@ public CatalogEntry fetchAndUpdateService(String serviceId, CatalogEntry service
         kernelService.supported_locales = service.getSupportedLocales();
         kernelService.geographical_areas = service.getGeographicalAreas();
         kernelService.restricted_areas = service.getRestrictedAreas();
-        kernelService.visible = service.isVisible();
+        kernelService.otherProperties.replace("visible", null);
```

Comment:
I may have misread something, but if `"visible"` is added to `BLACKLISTED`, then it will never be in `otherProperties` and this line would be useless/unnecessary.

(and `"restricted"` needs to be handled the same way)